test/EventStore.Client.Tests.Common/EventStoreTestServer.cs

```diff
@@ -7,17 +7,42 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Ductus.FluentDocker.Builders;
+using Ductus.FluentDocker.Extensions;
 using Ductus.FluentDocker.Model.Builders;
 using Ductus.FluentDocker.Services;
+using Ductus.FluentDocker.Services.Extensions;
 using Polly;
 
-#nullable enable
 namespace EventStore.Client {
 	public class EventStoreTestServer : IEventStoreTestServer {
 		private readonly string _hostCertificatePath;
 		private readonly IContainerService _eventStore;
 		private readonly HttpClient _httpClient;
 		private static readonly string ContainerName = "es-client-dotnet-test";
+		private static readonly string DockerImage = $"ghcr.io/eventstore/eventstore:{GlobalEnvironment.ImageTag}";
+
+		private static Version? _version;
+		public static Version? Version => _version ??= GetVersion();
+
+		private static Version? GetVersion() {
+			const string versionPrefix = "EventStoreDB version";
+
+			using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+			using var eventstore = new Builder().UseContainer()
+				.UseImage(DockerImage)
+				.Command("--version")
+				.Build()
+				.Start();
+			using var log = eventstore.Logs(true, cts.Token);
+			foreach (var line in log.ReadToEnd()) {
+				if (line.StartsWith(versionPrefix) &&
+				    Version.TryParse(line[(versionPrefix.Length + 1)..].Split(' ')[0], out var version)) {
+					return version;
+				}
+			}
+
+			return null;
```

Comment:
If (somehow) there is a version read from the container that we couldn't parse, we'll return null here and then accessing the `Version` property will keep rerunning this container. Perhaps want to avoid the retries in this case

Reply:
We'll throw an exception instead